script/urlChecker.ts

```diff
@@ -0,0 +1,93 @@
+import fetch from 'node-fetch';
+import ProgressBar from 'progress';
+import { argv } from 'yargs';
+import { content as contentRoute } from '../src/app/content/routes';
+import { Book, BookWithOSWebData, LinkedArchiveTreeSection } from '../src/app/content/types';
+import { findTreePages } from '../src/app/content/utils/archiveTreeUtils';
+import { getBookPageUrlAndParams, getUrlParamForPageId } from '../src/app/content/utils/urlUtils';
+import { assertDefined } from '../src/app/utils';
+import { findBooks } from './utils/bookUtils';
+
+const {
+  rootUrl,
+  bookId,
+  bookVersion,
+  archiveUrl,
+  useUnversionedUrls,
+} = argv as {
+  rootUrl?: string;
+  bookId?: string;
+  bookVersion?: string;
+  archiveUrl?: string;
+  useUnversionedUrls?: boolean;
+};
+
+async function checkPages(bookSlug: string, pages: string[]) {
+  let anyFailures = false;
+  const bar = new ProgressBar(`checking ${bookSlug} [:bar] :current/:total (:etas ETA)`, {
+    complete: '=',
+    incomplete: ' ',
+    total: pages.length,
+    width: 20,
+  });
+
+  const notFound: string[] = [];
+
+  for (const pageUrl of pages) {
+    try {
+      const response = await fetch(`${rootUrl}${pageUrl}`);
+      if (response.status === 404) {
+        notFound.push(pageUrl);
+      }
+    } catch (e) {
+      anyFailures = true;
+      bar.interrupt(`- (error loading) ${pageUrl}`);
+    }
+
+    bar.tick();
+  }
```

Comment:
This will be checking one url at the time. You can use `Promise.all` to make it faster.
I don't know how many urls there is to check, but if there is like hundreds of them then we can split them into blocks for ~50
```
const promises: () => Promise<void>[] = []

  for (const pageUrl of pages) {
    promises.push(() => {
      fetch(`${rootUrl}${pageUrl}`)
      .then(r => {
        if (r.status === 404) { notFound.push(pageUrl); }
        bar.tick();
      })
      .catch(e => {
        anyFailures = true;
        bar.interrupt(`- (error loading) ${pageUrl}`);
        bar.tick();
      })
    })
  }
  
  await Promise.all(promises.map(p => p()))
```